src/api/layers/GPXLayer.class.js

```diff
@@ -1,37 +1,53 @@
 
 export default class GPXLayer extends AbstractLayer {
     /**
-     * @param {string} gpxFileUrl URL to the GPX file (can also be a local file URI)
-     * @param {boolean | null} [visible=true] If the layer is visible on the map (or hidden). When
-     *   `null` is given, then it uses the default value. Default is `true`
-     * @param {number | null} [opacity=1.0] The opacity of this layer, between 0.0 (transparent) and
-     *   1.0 (opaque). When `null` is given, then it uses the default value. Default is `1.0`
-     * @param {string | null} [gpxData=null] Data/content of the GPX file, as a string. Default is
-     *   `null`
-     * @param {object | null} [gpxMetadata=null] Metadata of the GPX file. This object contains all
-     *   the metadata found in the file itself within the <metadata> tag. Default is `null`
+     * @param {String} gpxLayerData.gpxFileUrl URL to the GPX file (can also be a local file URI)
+     * @param {Boolean} [gpxLayerData.visible=true] If the layer is visible on the map (or hidden).
+     *   Default is `true`
+     * @param {Number} [gpxLayerData.opacity=1.0] The opacity of this layer, between 0.0
+     *   (transparent) and 1.0 (opaque). Default is `1.0`
+     * @param {String | null} [gpxLayerData.gpxData=null] Data/content of the GPX file, as a string.
+     *   Default is `null`
+     * @param {GPXMetadata | null} [gpxLayerData.gpxMetadata=null] Metadata of the GPX file. This
+     *   object contains all the metadata found in the file itself within the <metadata> tag.
+     *   Default is `null`
+     * @throws InvalidLayerDataError if no `gpxLayerData` is given or if it is invalid
      */
-    constructor(gpxFileUrl, visible = null, opacity = null, gpxData = null, gpxMetadata = null) {
+    constructor(gpxLayerData) {
+        if (!gpxLayerData) {
+            throw new InvalidLayerDataError('Missing GPX layer data', gpxLayerData)
+        }
+        const {
+            gpxFileUrl = null,
+            visible = true,
+            opacity = 1.0,
+            gpxData = null,
+            gpxMetadata = null,
+        } = gpxLayerData
+        if (gpxFileUrl === null) {
+            throw new InvalidLayerDataError('Missing GPX file URL', gpxLayerData)
+        }
         const isLocalFile = !gpxFileUrl.startsWith('http')
         const attributionName = isLocalFile ? gpxFileUrl : new URL(gpxFileUrl).hostname
-        super(
-            gpxMetadata?.name ?? 'GPX',
-            LayerTypes.GPX,
-            opacity ?? 1.0,
-            visible ?? true,
-            [new LayerAttribution(attributionName)],
-            false, //hasTooltip
-            true // isExternal
-        )
+        super({
+            name: gpxMetadata?.name ?? 'GPX',
+            id: `GPX|${gpxFileUrl}`,
+            type: LayerTypes.GPX,
+            opacity: opacity,
+            visible: visible,
+            attributions: [new LayerAttribution(attributionName)],
+            hasTooltip: false,
```

Comment:
Probably not for this PR, but at the end of the day we will also display the tooltips for GPX, am I right ? We could simply display the trackname and the coordinates of the feature.